sites/partners/src/listings/PaperListingForm/sections/LotteryResults.tsx

```diff
@@ -0,0 +1,199 @@
+import React from "react"
+import { useFormContext } from "react-hook-form"
+import {
+  cloudinaryUrlFromId,
+  t,
+  AppearanceBorderType,
+  AppearanceStyleType,
+  Button,
+  CloudinaryUpload,
+  Drawer,
+  Dropzone,
+  MinimalTable,
+  TableThumbnail,
+} from "@bloom-housing/ui-components"
+import {
+  AssetsService,
+  ListingEvent,
+  ListingEventCreate,
+  ListingEventType,
+} from "@bloom-housing/backend-core/types"
+
+interface LotteryResultsProps {
+  submitCallback: (data: { events: ListingEvent[] }) => void
+  drawerState: boolean
+  showDrawer: (toggle: boolean) => void
+}
+
+const LotteryResults = (props: LotteryResultsProps) => {
+  const formMethods = useFormContext()
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { watch } = formMethods
+
+  const cloudName = process.env.cloudinaryCloudName
+  const uploadPreset = process.env.cloudinarySignedPreset
+
+  const { submitCallback, drawerState, showDrawer } = props
+  const [progressValue, setProgressValue] = React.useState(0)
+  const [cloudinaryData, setCloudinaryData] = React.useState({
+    id: "",
+    url: "",
+  })
+
+  const listingEvents = watch("events")
+
+  const resetDrawerState = () => {
+    showDrawer(false)
+  }
+
+  const savePDF = () => {
+    const updatedEvents = [...listingEvents]
+
+    const lotteryIndex = updatedEvents.findIndex(
+      (event) => event.type === ListingEventType.lotteryResults
+    )
+    if (lotteryIndex > -1) {
+      updatedEvents.splice(lotteryIndex, 1)
+    }
+
+    const newEvent: ListingEventCreate = {
+      type: ListingEventType.lotteryResults,
+      startTime: new Date(),
+      file: {
+        fileId: cloudinaryData.id,
+        label: "cloudinaryPDF",
+      },
+    }
+
+    updatedEvents.push(newEvent as ListingEvent)
+    submitCallback({ events: updatedEvents })
+  }
+
+  const resultsTableHeaders = {
+    preview: "t.preview",
+    fileName: "t.fileName",
+    actions: "",
+  }
+
+  /*
+    Show a preview of the uploaded file within the photo drawer
+  */
+  const previewTableRows = []
+  if (cloudinaryData.url !== "") {
+    previewTableRows.push({
+      preview: (
+        <TableThumbnail>
+          <img alt="PDF preview" src={cloudinaryData.url} />
+        </TableThumbnail>
+      ),
+      fileName: cloudinaryData.id.split("/").slice(-1).join(),
+      actions: (
+        <Button
+          type="button"
+          className="font-semibold uppercase text-red-700"
+          onClick={() => {
+            setCloudinaryData({
+              id: "",
+              url: "",
+            })
+            setProgressValue(0)
+          }}
+          unstyled
+        >
+          {t("t.delete")}
+        </Button>
+      ),
+    })
+  }
+
+  /*
+    Uploader callback for the dropzone
+  */
+  const fileUploader = async (file: File) => {
```

Comment:
Is this fileUploader function identical, or at least close enough to the function found in ListingPhoto to warrant a shared function?

Reply:
Good call, we can DRY that up.